docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,199 @@
+consumers.
+
+Non-goals
+~~~~~~~~~
+
+- Persisting newly added Grafana dashboards or new Grafana datasources
+  especially for modifications added via the Grafana UI cannot be stored in
+  ConfigMaps.
+  To handle this particular user case, we need to configure Grafana deployments
+  to use persistent volume claims and then document how to create volumes that
+  will handle the persistence of these settings across Pod restarts.
+
+- ConfigMaps as used in the above design for storing and persisting user
+  settings need periodic backups for recovery purposes. This implies that
+  tested backup and restore measures for the etcd database need to be put
+  in place to guarantee recovery in cases where there is a full site loss.
+
+Rejected design choices
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Consul KV vs ConfigMap
+~~~~~~~~~~~~~~~~~~~~~~
+
+This approach offers a full fledge KV store with a /kv endpoint which allows
+CRUD operations on all KV data stored in it.
+Consul KV also allows access to past versions of objects and has an optimistic
+concurrency when manipulating multiple objects.
+
+Note that, Consul KV store was rejected because managing operations such as
+performing full backups, system restores for a full fledged KV system
+requires time and much more efforts than the ConfigMap approach.
+
+Operator (Custom Controller) vs ConfigMap
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Operators are useful in that, they provide self-healing functionalities on a
+reactive basis. When a user changes a given configuration, it is easy to
+reconcile and apply these changes to the in-cluster objects.
+
+The Operator approach was rejected because it is much more complex, requires
+much more effort to realize and there is no real need for applying changes
+using this method because configuration changes are not frequent
+(for a typical MetalK8s admin, changing the number of replicas for a given
+deployment could happen once in 3 months) as such, having an operator watch for
+object changes is not significant and not very useful at this point in time.
+
+
+Implementation Details
+----------------------
+
+Storage format
+~~~~~~~~~~~~~~
+
+A sample ConfigMap can be defined with the following fields.
+
+An example of such a ConfigMap:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: <namespace>
+      name: <config-name>
+    data:
+      config.yaml: |-
+        apiVersion: <object-version>
+        kind: <kind>
+        spec:
+          <key>: <values>
+
+**Use case 1:**
+
+Configure and store the number of replicas for service specific Deployments
+found in the `metalk8s-monitoring` namespace using the ConfigMap format.
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: metalk8s-monitoring
+      name: metalk8s-grafana-userconfig
+    data:
+      config.yaml: |-
+        apiVersion: metalk8s.scality.com/v1alpha1
+        kind: GrafanaUserConfig
```

Comment:
Not sure I like the `GrafanaUserConfig` name. We're not configuring Grafana users here... `GrafanaCustomizations`? `GrafanaConfig`? ...

Reply:
Picked the later!